src/main/java/edu/texas/social_computing/hospitals/RHRP.java

```diff
@@ -0,0 +1,135 @@
+            if (currentResident.hasPartner()) {
+                Resident partner = residentTable.getResidentById(currentResident.getPartnerId());
+                // if either current or partner did not get matched, make sure they both become unmatched them and put them both in the queue
+                if (!m.hasAssignment(currentResident) || !m.hasAssignment(partner)) {
+                    m.unassign(currentResident);
+                    m.unassign(partner);
+                    freeResidents.add(currentResident);
+                    freeResidents.add(partner);
+                }
+                // if i dont have a partner check if im unmatched
+            } else {
+                if (!m.hasAssignment(currentResident)) {
+                    freeResidents.add(currentResident);
+                }
+            }
+        }
+        return m;
+    }
+
+    public static Matching run(HospitalTable hospitalTable,
+                               ResidentTable residentTable,
+                               Queue<Resident> freeResidents) {
+        return run(hospitalTable, residentTable, freeResidents, new Matching());
+    }
+
+    public static void main(String[] args) throws FileNotFoundException {
+        List<Resident> residents = FileImporter.importResidents(args[0]);
+        List<Hospital> hospitals = FileImporter.importHospitals(args[1]);
+        System.out.println("loaded Files");
+        System.out.println("residents: " + Integer.toString(residents.size()));
+        System.out.println("hospitals: " + Integer.toString(hospitals.size()));
+
+        // make lookup tables
+        HospitalTable hospitalTable = HospitalTable.create(hospitals);
+        ResidentTable residentTable = ResidentTable.create(residents);
+        System.out.println("generated tables");
+
+        // call HRP
+        Matching m = RHRP.run(hospitalTable, residentTable, new ArrayDeque<>(residentTable.getAll()));
+        System.out.println(m);
+
+
+        Set<String> unassignedIds = m.getAllUnassigned(residentTable.getAll()).stream()
+                .map(Resident::getId)
+                .collect(ImmutableSet.toImmutableSet());
+        System.out.println("Unassigned: " + unassignedIds);
+    }
+
+    private static void tryToMatch(Hospital h, Resident resident,
+                                   ResidentTable residentTable, Queue<Resident> freeResidents, Matching m) {
+
+        // Only if the hospital has this resident on its pref list can it accept the resident
+        if (h.getPreferences().contains(resident.getId())) {
+            m.assign(resident, h);
+
+            // if hospital now has too many residents
+            if (m.isOverSubscribed(h)) {
+                Resident worstResident = m.getWorstAssignedResident(h);
+
+                // unassign the worst resident then add them back to the queue
+                m.unassign(worstResident);
+                freeResidents.add(worstResident);
```

Comment:
What if the second partner kicks the first out of the hospital since he/she is better ranked according to the hospital? 
Would then the first partner get added to the free list?
Wouldn't line 69 add that "currentResident" to the free list again? Or is there a check/property to make sure you can't be added twice?